src/QMCWaveFunctions/BsplineFactory/SplineC2C.h

```diff
@@ -161,11 +161,11 @@ class SplineC2C : public BsplineSet
    */
   void assign_vgl_from_l(const PointType& r, ValueVector_t& psi, GradVector_t& dpsi, ValueVector_t& d2psi);
 
-  virtual void evaluateVGL(const ParticleSet& P,
```

Comment:
I'm wondering which tool suggested this change. It doesn't seem correct to me.
For example `HybridRepCplx<SplineC2C<double>>`
`HybridRepCplx->SplineC2C->BsplineSet->SPOSet` for evaluateVGL virtual function.

Reply:
clang-tidy-12
```
/home/svh/Documents/qmcpack/src/QMCWaveFunctions/BsplineFactory/SplineC2C.h:164:8: warning: annotate this function with 'override' or (rarely) 'final' [modernize-use-override]
  void evaluateVGL(const ParticleSet& P,
       ^
/home/svh/Documents/qmcpack/src/QMCWaveFunctions/BsplineFactory/SplineC2C.h:168:41: note: FIX-IT applied suggested code changes
                   ValueVector_t& d2psi);
                                        ^
```

Reply:
This function was marked `override` and still is. Clang-tidy just removed `virtual` since it is not required.

Reply:
I'm not quite getting why virtual is not required. Could you explain a bit more? I thought virtual is needed for the inheritance to propagate.

Reply:
`virtual` is only necessary in the base class.

```
If some member function vf is declared as virtual in a class Base, and some class Derived, which is derived, directly or indirectly, from Base, has a declaration for member function with the same
    name
    parameter type list (but not the return type)
    cv-qualifiers
    ref-qualifiers
Then this function in the class Derived is also virtual (whether or not the keyword virtual is used in its declaration) and overrides Base::vf (whether or not the word override is used in its declaration).
```
https://en.cppreference.com/w/cpp/language/virtual

```
virtual on non base class implementations was used to help indicate to the user that a function was virtual. C++ compilers did not use the presence of this to signify an overridden function.
```
https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html